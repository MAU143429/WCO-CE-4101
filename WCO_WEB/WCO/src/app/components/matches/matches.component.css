h1{
    text-align:center;
    position: absolute;
    margin: 0;
    font-size: 40px;
    color: #fff;
    z-index: 2;
    line-height: 83px;
}
.line-1 {
    height: 1px;
    background: black;
  }
  body {
    min-height: 100%;
    }
    body, div, form, input, select, textarea, label, p { 
    padding: 0;
    margin: 0;
    outline: none;
    font-family: Roboto, Arial, sans-serif;
    font-size: 16px;
    color: #666;
    line-height: 22px;
    text-align: center;
    }
    textarea {
    width: calc(100% - 12px);
    padding: 5px;
    }
    select {
      width: calc(52% - 12px);
      padding: 5px;
      }
    .testbox {
    display: flex;
    justify-content: center;
    align-items: center;
    height: inherit;
    padding: 20px;
    }
    form {
    width: 85%;
    padding: 20px;
    border-radius: 6px;
    background: #fff;
    box-shadow: 0 0 8px  #669999; 
    }
    .banner {
    position: relative;
    height: 300px;
    background-image: url("src/assets/images/matches-bg.jpg");  
    background-size: cover;
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
    }
    .banner::after {
    content: "";
    background-color: rgba(0, 0, 0, 0.2); 
    position: absolute;
    width: 100%;
    height: 100%;
    }
    input, select, textarea {
    margin-bottom: 10px;
    border: 1px solid #ccc;
    border-radius: 3px;
    }
    input {
    width: calc(50% - 15px);
    padding: 5px;
    align-items: center;
    text-align: center;
    }
    input[type="date"] {
    padding: 4px 5px;
    }
    textarea {
    width: calc(100% - 12px);
    padding: 5px;
    }
    .item:hover p, .item:hover i, .question:hover p, .question label:hover, input:hover::placeholder {
    color:  #669999;
    }
    .item input:hover, .item select:hover, .item textarea:hover {
    border: 1px solid transparent;
    box-shadow: 0 0 3px 0  #669999;
    color: #669999;
    }
    .item {
    position: relative;
    margin: 10px 0;
    }
    .item span {
    color: rgb(245, 59, 59);
    }
    .week {
    display:flex;
    justify-content:space-between;
    }
    .colums {
    display:flex;
    justify-content:space-between;
    flex-direction:row;
    flex-wrap:wrap;
    }
    .colums div {
    width:48%;
    }
    input[type="date"]::-webkit-inner-spin-button {
    display: none;
    }
    .item i, input[type="date"]::-webkit-calendar-picker-indicator {
    position: absolute;
    font-size: 20px;
    color:  #a3c2c2;
    }
    .item i {
    right: 1%;
    top: 30px;
    z-index: 1;
    }
    input[type=radio], input[type=checkbox]  {
    display: none;
    }
    label.radio {
    position: relative;
    display: inline-block;
    margin: 5px 20px 15px 0;
    cursor: pointer;
    }
    .question span {
    margin-left: 30px;
    }
    .question-answer label {
    display: block;
    }
    label.radio:before {
    content: "";
    position: absolute;
    left: 0;
    width: 17px;
    height: 17px;
    border-radius: 50%;
    border: 2px solid #ccc;
    }
    input[type=radio]:checked + label:before, label.radio:hover:before {
    border: 2px solid  #669999;
    }
    label.radio:after {
    content: "";
    position: absolute;
    top: 6px;
    left: 5px;
    width: 8px;
    height: 4px;
    border: 3px solid  #669999;
    border-top: none;
    border-right: none;
    transform: rotate(-45deg);
    opacity: 0;
    }
    input[type=radio]:checked + label:after {
    opacity: 1;
    }
    .flax {
    display:flex;
    justify-content:space-around;
    }
    .btn-block {
    margin-top: 10px;
    text-align: center;
    }
    button {
    width: 150px;
    padding: 10px;
    border: none;
    border-radius: 5px; 
    background:  #1277D9;
    font-size: 16px;
    color: #fff;
    cursor: pointer;
    }
    button:hover {
    background:  #a3c2c2;
    }
    @media (min-width: 568px) {
    .name-item, .city-item {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    }
    .name-item input, .name-item div {
    width: calc(50% - 20px);
    }
    .name-item div input {
    width:97%;}
    .name-item div label {
    display:block;
    padding-bottom:5px;
    }
    }
 /*   
  body{
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 10px;
    background-color: white;
  
  }
  .container{
    max-width: 1400px;
    width: 100%;
    background-color: rgba(255, 255, 255, 0.295);
    padding: 25px 30px;
    border-radius: 5px;
    box-shadow: 0 5px 10px rgba(0,0,0,0.20);
  }
  .container .title{
    font-size: 30px;
    font-weight: 500;
    text-align: center;
    position: relative;
  }
  .container .title::before{
    content: "";
    position: absolute;
    bottom: 0;
    height: 3px;
    width: 190px;
    border-radius: 5px;
    background: linear-gradient(135deg, #ffffff, #71aae4, #1277D9);
  }
  .content form .match-form{
    display: flex;
    flex-wrap: wrap;
    margin: 20px 0 12px 0;

  } 
  form .match-form div{
    width: 100vw;
    position: absolute;
  
    display: flex;
    flex-flow: column wrap;
    align-items: center;
   
  }

 form .input-sede span.details{
    display: block;
    font-weight: 500;
    margin-bottom: 5px;
    text-align: center;
  } 
  .matches-form .input-sede input.details{
    position: relative;
    display: block;
    height: 45px;
    width: 100%;
    font-size: 16px;
    border-radius: 5px;
    padding-left: 15px;
    border: 1px solid #ccc;
    border-bottom-width: 2px;
}
*/