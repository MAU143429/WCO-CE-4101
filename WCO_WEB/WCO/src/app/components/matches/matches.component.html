<html lang="en">
<head>
  <meta charset="utf-8">
  <title>WCO</title>
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="stylesheet" href="matches.component.css">
</head>
    <body>
        <header>
            <h2>WCO...</h2>
                
            <div class="line-1"></div>
        </header>
        <main>
            <div class="testbox">
                <form action="/">
                  <div class="banner">
                    <h1>Crear partido</h1>
                  </div>
                  <div class="colums">
                    <div class="item">
                        <h3>Torneo: Nombre del torneo</h3>
                    </div>
                    <div class="item">
                        <h3>Fase: Nombre de la fase</h3>
                    </div>
                    <div class="item">
                      <label for="sede">Sede del partido<span>*</span></label><br>
                      <input id="sede" type="text" name="sede" placeholder="Ingrese sede del partido" required/>
                    </div>
                    <div class="item">
                      <label for="inicio">Fecha y hora de inicio<span>*</span></label><br>
                      <input id="inicio" type="datetime-local" name="inicio" required/>
                    </div>
                    <div class="item">
                        <label for="equipo1">Equipo 1<span>*</span></label><br>
                        <select name="equipo1" id="equipo1" placeholder="Seleccione equipo 1">
                            <option value="1">Brasil</option>
                            <option value="2">Costa Rica</option>
                            <option value="3">Canad치</option>
                        </select>
                    </div>
                    <div class="item">
                        <label for="equipo2">Equipo 2<span>*</span></label><br>
                        <select name="equipo2" id="equipo2" placeholder="Seleccione equipo 2">
                            <option value="1">Croacia</option>
                            <option value="2">Espa침a</option>
                            <option value="3">Senegal</option>
                        </select>
                    
                    </div>
                    
                  </div>
                  <div class="btn-block">
                    <button type="submit" href="/">Agregar partido</button>
                  </div>
                </form>
            </div>
        </main>
        <footer>

        </footer>
        
</body>
</html>

<!-- 
            <div class="container">
             
                <div class="title">Crear partido</div>
                
                <div class="content">
                    <form action="#">
                        <div class="matches-form">
                            <div class="input-sede">
                                <span >Sede del partido <span style="color: rgb(245, 59, 59);">*</span> </span>
                                <input type="text"  placeholder="Ingrese la sede del torneo " required>
                            </div>
    
                            <div class="input-sede">
                                <span >Fecha y hora de inicio <span style="color: rgb(245, 59, 59);">*</span> </span>
                                <input type="datetime-local"  placeholder="Ingrese la hora de inicio del partido" required>
                            </div>
    
                            <div>
                                    <span >Equipo 1 <span style="color: rgb(245, 59, 59);">*</span> </span>
                                    <select name="team1" id="team1" placeholder="Seleccione equipo 1">
                                        <option value="1">Brasil</option>
                                        <option value="2">Costa Rica</option>
                                        <option value="3">Canad치</option>
                                    </select>
                            </div>
                            <div>
                                    <span>Equipo 2 <span style="color: rgb(245, 59, 59);">*</span> </span>
                                    <select name="team2" id="team2" placeholder="Seleccione equipo 2">
                                        <option value="1">Croacia</option>
                                        <option value="2">Espa침a</option>
                                        <option value="3">Senegal</option>
                                    </select>
                              
                            </div>
                        </div>
                        <div class="button">
                            <input type="submit" value="Crear partido"  routerLink="">
                        </div>
                    </form>
                </div>
            </div> -->