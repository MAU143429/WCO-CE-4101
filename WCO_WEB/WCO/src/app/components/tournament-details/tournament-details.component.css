body{
  height: 100vh;
  justify-content: center;
  align-items: center;
  padding: 10px;
  background-color: white;

}
.container{
  max-width: 1700px;
  width: 100%;
  background-color: rgba(255, 255, 255, 0.295);
  padding: 25px 30px;
  border-radius: 5px;
  box-shadow: 0 5px 10px rgba(0,0,0,0.20);
}
.title{
  font-size: 30px;
  font-weight: 500;
  margin-left: 5%;
  margin-top: 1%;
  position: relative;
}
.subtitle{
  font-size: 20px;
  font-weight: 300;
  margin-left: 5%;
  position: relative;
}
.bracket-title{
  font-size: 25px;
  font-weight: 300;
}

.create-match-btn{
  background-color: transparent;
  border: none;
}
