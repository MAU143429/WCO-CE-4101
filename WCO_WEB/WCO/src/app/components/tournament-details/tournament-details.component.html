<!DOCTYPE html>
<html lang="en" dir="ltr">
  <head>
    <meta charset="UTF-8">
    <!---Referencia al archivo css que contiene el estilo--->
    <link rel="stylesheet" href="tournament-details.component.css">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
   </head>
<!---Componentes que conforman la seccion de ver los detalles de un torneo--->

<body>
  <app-navbar></app-navbar>
  <br>
  <br>
  <br>
  <div *ngFor="let data of tournamentsData">
    <div class="title">{{data.name}}</div>
    <div class="subtitle">{{data.startDate}} - {{data.endDate}} &nbsp;&nbsp;&nbsp;&nbsp;   Tipo de torneo: {{data.type}}</div>
    <br>
  </div>
  <div class="container">
    <div class="content" >
      <ngb-accordion  #acc="ngbAccordion" [closeOthers]="true" *ngFor="let bracket of matchData; index as i">
        <ngb-panel>
          <ng-template ngbPanelTitle class="bracket-title">
            <span>{{bracketsData[i].name}}</span>
          </ng-template>
          <ng-template  ngbPanelContent>
            <div *ngFor="let match of bracket">
              <span (click)="viewMatch(match.mId)">{{match.teams[0].name}} &nbsp;&nbsp;&nbsp;&nbsp; VS&nbsp;&nbsp;&nbsp;&nbsp; {{match.teams[1].name}}&nbsp;&nbsp;&nbsp;&nbsp;  Sede: {{match.venue}} &nbsp;&nbsp;&nbsp;&nbsp;Horario: {{match.date}} {{match.startTime}}</span>
            </div>
            <input class="create-match-btn" type="button"  value="+ Agregar Partido" (click)="redirectMatch(bracketsData[i])">
          </ng-template>
        </ngb-panel>
      </ngb-accordion>
    </div>
  </div>
</body>
</html>
