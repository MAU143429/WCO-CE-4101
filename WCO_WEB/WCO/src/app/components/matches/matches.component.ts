import { MatchesService } from './../../services/matches.service';
import { Match } from './../../model/match';
import { Component, OnInit, NgModule } from '@angular/core';
import { Router } from '@angular/router';
import { JsonPipe } from '@angular/common';
@Component({
  selector: 'app-matches',
  template: `
  Match
`,
  templateUrl: './matches.component.html',
  styleUrls: ['./matches.component.css']
})
export class MatchesComponent implements OnInit {
  newMatch: Match = new Match
  //selectionsList: 
  constructor(private game:MatchesService) {  }

  ngOnInit(): void {

  }

  insertMatch(newMatch: Match){
    console.log(JSON.stringify(newMatch))
    this.game.addNewMatch(newMatch).subscribe((match)=>
    console.log(this.newMatch)
    )
  }

}
