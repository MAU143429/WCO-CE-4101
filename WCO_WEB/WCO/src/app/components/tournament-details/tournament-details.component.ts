import { Component, Input, OnInit } from '@angular/core';
import { TournamentService } from '../../services/tournament.service';
import { Tournaments } from '../../interface/tournaments';
import { InternalService } from '../../services/internal.service';
import { BracketName } from '../../interface/bracket-name';
import { DbBracket } from 'src/app/model/db-bracket';
import { Router } from '@angular/router';
import {
  NgbPanelChangeEvent,
  NgbPanelToggle,
} from '@ng-bootstrap/ng-bootstrap';
import { MatchesService } from 'src/app/services/matches.service';
import { DbMatch } from 'src/app/interface/db-match';

@Component({
  selector: 'app-tournament-details',
  templateUrl: './tournament-details.component.html',
  styleUrls: ['./tournament-details.component.css'],
})
export class TournamentDetailsComponent implements OnInit {
  tournamentID: string = '';
  tournamentsData: Tournaments[] = [];
  bracketsData: DbBracket[] = [];
  matchData: any[] = [];
  bracketMatches: DbMatch[] = [];

  /**
   * Este metodo permite realizar un peque√±o delay
   * @param ms el tiempo del delay en ms
   */
  async delay(ms: number) {
    await new Promise<void>((resolve) => setTimeout(() => resolve(), ms)).then(
      () => console.log('fired')
    );
  }

  constructor(
    private service: TournamentService,
    private matchService: MatchesService,
    private connection: InternalService,
    private router: Router
  ) {}

  ngOnInit(): void {
    this.connection.tournamentId.subscribe(
      (data) => (this.tournamentID = data)
    );

    this.delay(50).then(() => {
      this.service
        .getTournamentbyID(this.tournamentID)
        .subscribe((data) => (this.tournamentsData = data));
    });

    this.delay(100).then(() => {
      this.bracketsData = this.tournamentsData[0].brackets;
      this.matchService
        .getMatchesByBracketId(this.tournamentsData[0].toId)
        .subscribe((data) => (this.matchData = data));
    });
  }

  /**
   * Este metodo permite enviar la informacion sobre el bracket en donde se encuentra
   * el partido almacenado que se desea abrir.
   * @param bracket bracket donde esta el partido
   */
  redirectMatch(bracket: any) {
    console.log('SOY BRACKET', bracket);
    this.connection.setCurrentBracket(bracket);
    this.delay(100).then(() => {
      this.router.navigate(['/create-match']);
    });
  }

  viewMatch(match: any) {
    this.connection.setCurrentMatch(match);
    this.delay(100).then(() => {
      this.router.navigate(['/match-details']);
    });
  }
}
