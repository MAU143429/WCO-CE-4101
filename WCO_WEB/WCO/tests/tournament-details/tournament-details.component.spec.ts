import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { http } from '@mobiscroll/angular/dist/js/core/util/http';
import { Observable, of } from 'rxjs';
import { TournamentService } from '../../src/app/services/tournament.service';
import { MatchesService } from '../../src/app/services/matches.service';

import { TournamentDetailsComponent } from '../../src/app/components/tournament-details/tournament-details.component';
import { NgbAccordion, NgbAccordionModule, NgbNavModule } from '@ng-bootstrap/ng-bootstrap';

const dummyTournament = [{
  toId: '1',
  name: 'Mundial',
  startDate: '10/10/2022',
  endDate: '11/11/2022',
  type: 'Selecciones',
  description: 'Muy chiva',
  teams: [{teId: 0, name: 'Team1'}, {teId: 1, name: 'Team2'}],
  brackets: [{  
    bId: 0,
    name: 'Final',
    tounamentId: 'A2X'}]
}]

const dummyMatch = [{
  startTime: '1:00',
  date: '11/15/2022',
  venue: 'Guapiles',
  bracketId: '0',
  idTeam1: 0,
  idTeam2: 1
}]

const mockTournamentService: {
  getTournamentbyID: () => Observable<any>
} = {
  getTournamentbyID: () => of(dummyTournament)
};

const mockMatchesService: {
  getMatchesByBracketId: () => Observable<any>
} = {
  getMatchesByBracketId: () => of(dummyMatch)
};

describe('TournamentDetailsComponent', () => {
  let component: TournamentDetailsComponent;
  let fixture: ComponentFixture<TournamentDetailsComponent>;
  let compiled: HTMLElement;
  let httpMock: HttpTestingController;
  let serviceT: TournamentService;
  let serviceM: MatchesService;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ TournamentDetailsComponent ],
      imports: [
        HttpClientTestingModule,
        NgbNavModule,
        NgbAccordionModule
      ],
      providers: [
        {provide: TournamentService, useValue: mockTournamentService},
        {provide: MatchesService, useValue: mockMatchesService}
      ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(TournamentDetailsComponent);
    component = fixture.componentInstance;
    serviceT = TestBed.inject( TournamentService );
    serviceM = TestBed.inject( MatchesService );

    httpMock = TestBed.inject( HttpTestingController );
    compiled = fixture.nativeElement;
    fixture.detectChanges();
  })

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe de hace match con el snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });

  test('En init debe traer torneo por id y partidos por id de bracket', () => {
    const getTournamentSpy = jest.spyOn(mockTournamentService, 'getTournamentbyID')
    getTournamentSpy.mockReturnValue(of(dummyTournament));

    const getMatchesSpy = jest.spyOn(mockMatchesService, 'getMatchesByBracketId')
    getMatchesSpy.mockReturnValue(of(dummyMatch));
    
    component.ngOnInit();
    expect(component.tournamentsData == dummyTournament)
    expect(component.matchData == dummyMatch)
  })

  test('debe actualizarse el valor de bracket actual usando redirectMatch', () => {
    const dummyBracket = 
    {  
      bId: 0,
      name: 'Final',
      tounamentId: 'A2X'
    }
    component.redirectMatch(dummyBracket);
    expect(localStorage.getItem("currentBracket"))
      .toEqual(dummyBracket.name)
  })

  test('debe actualizarse el valor de partido actual usando viewMatch', () => {

    component.viewMatch(JSON.stringify(dummyMatch[0]));
    const n = localStorage.getItem("currentMatch");
    expect(JSON.parse(n!))
      .toEqual(dummyMatch[0])
  })

});
