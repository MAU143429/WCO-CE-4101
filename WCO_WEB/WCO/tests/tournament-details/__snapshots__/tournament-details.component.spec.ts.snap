// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TournamentDetailsComponent debe de hace match con el snapshot 1`] = `
"<html lang=\\"en\\" dir=\\"ltr\\"><head><meta charset=\\"UTF-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"></head><body><app-navbar></app-navbar><br><br><br><div><div class=\\"title\\">Mundial</div><div class=\\"subtitle\\">10/10/2022 - 11/11/2022 &nbsp;&nbsp;&nbsp;&nbsp; Tipo de torneo: Selecciones</div><br></div><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"[object Object]\\"
}--><div class=\\"container\\"><div class=\\"content\\"><ngb-accordion role=\\"tablist\\" class=\\"accordion\\" ng-reflect-close-other-panels=\\"true\\" aria-multiselectable=\\"false\\"><!--container--><div class=\\"accordion-item\\"><div role=\\"tab\\" class=\\"accordion-header\\" id=\\"ngb-panel-1-header\\"><button type=\\"button\\" class=\\"accordion-button collapsed\\" ng-reflect-ngb-panel-toggle=\\"[object Object]\\" aria-expanded=\\"false\\" aria-controls=\\"ngb-panel-1\\"> <span>Final</span><!--bindings={
  \\"ng-reflect-ng-template-outlet\\": \\"[object Object]\\"
}--></button><!--bindings={
  \\"ng-reflect-ng-template-outlet-context\\": \\"[object Object]\\"
}--></div><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></div><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"[object Object]\\"
}--></ngb-accordion><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"[object Object]\\"
}--></div></div></body></html>"
`;
